File: _includes/code/2010-01-19/shift.rb
def shift(no, s) 
  [no << (s - 1), (no << s) / 2]
end
shift(2, 8) # => [256, 256]
shift(2, 1) # => [2, 2]
shift(2, 0) # => [1, 1]
shift(256, -4) # => [8, 8]

File: _includes/code/2010-01-19/shift.c
#include <stdio.h>
void shift(int no, int s) {
  int one = no << (s - 1);
  int two = (no << s) / 2;
  printf("Shift(%d, %d): %d, %d\n", no, s, one, two);
}
int main() {
  shift(2, 8); // 256, 256
  shift(2, 1); // 2, 2
  shift(2, 0); // 0, 1
  shift(256, -4); // 0, 0
  return 0;
}